src/H5FDint.c

```diff
@@ -2501,9 +2540,13 @@ H5FD_driver_query(const H5FD_class_t *driver, unsigned long *flags /*out*/)
 
     /* Check for the driver to query and then query it */
     if (driver->query)
-        ret_value = (driver->query)(NULL, flags);
-    else
-        *flags = 0;
+        /* Prepare & restore library for user callback */
+        H5_BEFORE_USER_CB_NOERR(FAIL)
```

Comment:
I think this should be contained in a block after the `if`.

Reply:
No, the H5_BEFORE_USER_CB_NOERR / H5_AFTER_USER_CB macros open and close a block in the macro definition.  Similar to FUNC_ENTER / FUNC_LEAVE.

Reply:
I actually agree with Matt here. Of course it works without it, but the curly braces make it more legible as it looks more like standard C.

Reply:
Seemed cleaner to me this way, but I'm not incredibly attached to it.   I'll make a pass through the code and add more braces.

Reply:
Done